### PR TITLE
apply clang-tidy modernize

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

Apply clang modernize:
modernize-use-override
modernize-avoid-bind
modernize-redundant-void-arg
modernize-raw-string-literal
modernize-deprecated-headers
modernize-make-unique
modernize-concat-nested-namespaces
modernize-deprecated-ios-base-aliases
modernize-make-shared
modernize-replace-auto-ptr
modernize-return-braced-init-list
modernize-shrink-to-fit
modernize-unary-static-assert
modernize-use-bool-literals
modernize-use-uncaught-exceptions

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Review by timosachsenberg — COMMENTED** [review]
I reviewed the changes by clang-modernize myself

**Comment by timosachsenberg** [comment]
modernize.equals-default resulted in some wrong changes. I reverted those

File: src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp [hunk 1]
```diff
@@ -627,15 +627,15 @@ namespace OpenMS::Internal
           os << "    <Peptide id=\"" << writeXMLEscape(it->id) << "\" sequence=\"" << it->sequence << "\">" << "\n";
           if (it->hasCharge())
           {
-            os << "      <cvParam cvRef=\"MS\" accession=\"MS:1000041\" name=\"charge state\" value=\"" <<  it->getChargeState() << "\"/>\n";
+            os << R"(      <cvParam cvRef="MS" accession="MS:1000041" name="charge state" value=")" <<  it->getChargeState() << "\"/>\n";
```
Comment (poshul) [inline src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp:630]: Is there a reason not to "rawify" the  "\"/>\n" ?

File: src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp [hunk 2]
```diff
@@ -783,7 +783,7 @@ namespace OpenMS::Internal
           // Precursor occurs exactly once (is required according to schema).
           // CV term MS:1000827 MUST be supplied for the TransitionList path
           os << "      <Precursor>" << "\n";
-          os << "        <cvParam cvRef=\"MS\" accession=\"MS:1000827\" name=\"isolation window target m/z\" value=\"" <<
+          os << R"(        <cvParam cvRef="MS" accession="MS:1000827" name="isolation window target m/z" value=")" <<
             precisionWrapper(it->getPrecursorMZ()) << "\" unitCvRef=\"MS\" unitAccession=\"MS:1000040\" unitName=\"m/z\"/>\n";
```
Comment (poshul) [inline src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp:787]: Same here, can this be converted to raw

File: src/openms/source/FORMAT/PepXMLFile.cpp [hunk 3]
```diff
@@ -437,9 +437,9 @@ namespace OpenMS
       replace(base_name.begin(), base_name.end(), '.', '_');
     }
 
-    f << "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" << "\n";
-    f << "<msms_pipeline_analysis date=\"2007-12-05T17:49:46\" xmlns=\"http://regis-web.systemsbiology.net/pepXML\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://sashimi.sourceforge.net/schema_revision/pepXML/pepXML_v117.xsd\" summary_xml=\".xml\">" << "\n";
-    f << "<msms_run_summary base_name=\"" << base_name << "\" raw_data_type=\"raw\" raw_data=\"." << raw_data << "\" search_engine=\"" << search_engine_name << "\">" << "\n";
+    f << R"(<?xml version="1.0" encoding="UTF-8"?>)" << "\n";
+    f << R"(<msms_pipeline_analysis date="2007-12-05T17:49:46" xmlns="http://regis-web.systemsbiology.net/pepXML" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://sashimi.sourceforge.net/schema_revision/pepXML/pepXML_v117.xsd" summary_xml=".xml">)" << "\n";
+    f << "<msms_run_summary base_name=\"" << base_name << R"(" raw_data_type="raw" raw_data=".)" << raw_data << "\" search_engine=\"" << search_engine_name << "\">" << "\n";
```
Comment (poshul) [inline src/openms/source/FORMAT/PepXMLFile.cpp:442]: still some escaped characters have escaped

File: src/openms/source/FORMAT/IdXMLFile.cpp [hunk 4]
```diff
@@ -1085,7 +1085,7 @@ namespace OpenMS
     PeptideHit::PeakAnnotation::writePeakAnnotationsString_(val, annotations);
     if (!val.empty())
     {
-      os << String(indent, '\t') << "<" << writeXMLEscape(tag_name) << " type=\"string\" name=\"fragment_annotation\" value=\"" << writeXMLEscape(val) << "\"/>" << "\n";
+      os << String(indent, '\t') << "<" << writeXMLEscape(tag_name) << R"( type="string" name="fragment_annotation" value=")" << writeXMLEscape(val) << "\"/>" << "\n";
```
Comment (poshul) [inline src/openms/source/FORMAT/IdXMLFile.cpp:1088]: string could be more raw

File: src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp (hunk 1 above)
Comment (timosachsenberg) [reply src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp:630]: guess it's just awful to read with the newline and they decided against it?

File: src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp (hunk 2 above)
Comment (timosachsenberg) [reply src/openms/source/FORMAT/HANDLERS/TraMLHandler.cpp:787]: not sure why clang-tidy didn't do it

File: src/openms/source/FORMAT/IdXMLFile.cpp (hunk 4 above)
Comment (timosachsenberg) [reply src/openms/source/FORMAT/IdXMLFile.cpp:1088]: I guess it doesn't do that for short strings

File: src/openms/source/FORMAT/PepXMLFile.cpp (hunk 3 above)
Comment (timosachsenberg) [reply src/openms/source/FORMAT/PepXMLFile.cpp:442]: let them go until clang-tidy catches up

**Review by poshul — APPROVED** [review]

File: src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringShiftSuperimposer.cpp [hunk 5]
```diff
@@ -322,7 +322,7 @@ namespace OpenMS
 
         } // k
       } // i
-    } while (0);   // end of hashing (the extra syntax helps with code folding in eclipse!)
+    } while (false);   // end of hashing (the extra syntax helps with code folding in eclipse!)
```
Comment (hroest) [inline src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringShiftSuperimposer.cpp:325]: @timosachsenberg maybe we should remove extra control elements that only help with cosmetics?

Comment (timosachsenberg) [reply src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringShiftSuperimposer.cpp:325]: agree... this was not caught by clang-tidy but I only applied a few fixes so far. I can check if such nonsense can be removed automatically.

**Pull request merged**